base/array.jl

```diff
@@ -827,32 +827,36 @@ function findmax(a)
     if isempty(a)
         throw(ArgumentError("collection must be non-empty"))
     end
-    m = a[1]
-    mi = 1
-    for i=2:length(a)
-        ai = a[i]
+    i = start(a)
+    mi = i
+    m, i = next(a, i)
+    while !done(a, i)
+        iold = i
+        ai, i = next(a, i)
         if ai > m || m!=m
             m = ai
-            mi = i
+            mi = iold
         end
     end
-    return (m, mi)
+    return (m, iterstate(mi))
```

Comment:
This (and its `findmin` companion) is the breaking change, because for `LinearSlow` arrays the returned index will be a `CartesianIndex`. For 0.5 this is probably the way to go. Assuming we don't want to change it for 0.4, a backport-worthy solution would be to increment a counter in the loop and return that instead.